backend/cfg_selectgen.ml

```diff
@@ -1085,8 +1081,19 @@ module Make (Target : Cfg_selectgen_target_intf.S) = struct
     assert (Sub_cfg.exit_has_never_terminator sub_cfg);
     let sub_handlers =
       List.map
-        (fun ((_, _, _, label), (_, sub_handler)) ->
-          Sub_cfg.add_empty_block_at_start sub_handler ~label;
+        (fun ((rs, label), (_, sub_handler)) ->
+          (match flag with
+          | Normal | Recursive ->
+            Sub_cfg.add_empty_block_at_start sub_handler ~label
+          | Exn_handler ->
+            Sub_cfg.mark_as_trap_handler sub_handler ~exn_label:label;
```

Comment:
This block can be shared with the tail case, I think?

Reply:
Yes, I have done that now.